### PR TITLE
Adding an option to limit the number of sphinx processes.

### LINKED ISSUE #70712: Decrease documentation build load
When building documentation, the sphinx process uses as many resources as possible. 
This is reasonable for CI, but I would like a setting with a lighter load.

### PR DESCRIPTION
**Pull request by soburi**
If we build with the default settings, sphinx will be run with `-j auto` specified, which will generate the same number of processes as the number of processors.
Depending on the environment, this can be a very high load.

Introduces the variable `SPHINX_PROCESSES` to specify the number of parallels so that it can be specified at the command line execution.

Such like
```
make SPHINX_PROCESSES=1
```

Fix #70712

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
this needs to be available from CMake

File: doc/Makefile [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 BUILDDIR ?= _build
 DOC_TAG ?= development
-SPHINXOPTS ?= -j auto -W --keep-going -T
+SPHINX_PROCESSES ?= auto
```
Comment (kartben) [inline doc/Makefile:7]: nit: `SPHINX_JOBS` (`JOBS` would probably be fine too IMO)

**Review by kartben — COMMENTED** [review]
what Gerard said, + a suggestion to rename the option
Thanks for taking the time for sending this PR, I think many people will appreciate it :)

**Comment by jukkar** [comment]
It would be great if this gets merged, currently it is almost impossible to build docs locally with my laptop.

File: doc/Makefile (hunk 1 above)
Comment (soburi) [reply doc/Makefile]: Thank you for the suggestion, and I applied it.

**Comment by soburi** [comment]
@gmarull 

> this needs to be available from CMake

I moved the settings to the CMake side.
Additionally, I changed to use the default value defined in CMakeLists.txt instead of redefining the defaults in the Makefile.
(Not set -D... option if no needed)

**Review by gmarull — DISMISSED** [review]

**Comment by soburi** [comment]
@kartben 
Please check again.

**Comment by henrikbrixandersen** [comment]
I normally just use `make SPHINXOPTS_EXTRA=-j1 html-fast` or the like.

**Comment by soburi** [comment]
@henrikbrixandersen 

Thank you for commenting.

roger that.
Looking at the conversation here, it seems that there are many people who don't know this, so I've written it as a comment.

Since I'm deleting duplicates in Makefile and CMakeLists.txt in this PR, I would like to take this opportunity to cleanup it.

**Comment by MaureenHelm** [comment]
@kartben please revisit

**Pull request closed without merging**